glslc/src/file_includer.cc

```diff
@@ -18,17 +18,39 @@
 
 namespace glslc {
 
-std::pair<std::string, std::string> FileIncluder::include_delegate(
-    const char* filename) const {
-  const std::string full_path = file_finder_.FindReadableFilepath(filename);
-  std::vector<char> content;
-  if (!full_path.empty() && shaderc_util::ReadFile(full_path, &content)) {
-    return std::make_pair(full_path,
-                          std::string(content.begin(), content.end()));
+shaderc_includer_response* FileIncluder::GetInclude(const char* filename) {
+  file_full_path_ = file_finder_.FindReadableFilepath(filename);
+  if (!file_full_path_.empty() &&
+      shaderc_util::ReadFile(file_full_path_, &file_content_)) {
+    response_ = {
+        file_full_path_.c_str(), file_full_path_.length(), file_content_.data(),
+        file_content_.size(),
+    };
   } else {
-    // TODO(deki): Emit error message.
-    return std::make_pair("", "");
+    // The file to be included is not found or can not be opened. Response to be
+    // returned should have an empty string as path, and an error message as content.
+    char error_msg[] = "Cannot open or find include file.";
+    file_content_.insert(file_content_.begin(), error_msg,
```

Comment:
Use `vector::assign` instead of `insert`.

Use `std::begin(error_msg)` and `std::end(error_msg)` for better readability.